akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -166,7 +174,10 @@ private[akka] class TLSActor(settings: ActorMaterializerSettings,
       case Some(TLSClientAuth.Need) ⇒ engine.setNeedClientAuth(true)
       case _                        ⇒ // do nothing
     }
-    params.sslParameters foreach (p ⇒ engine.setSSLParameters(p))
+    params.sslParameters foreach { p ⇒
+      hostInfo foreach { case (host, _) ⇒ applySNI(host, engine, p) }
```

Comment:
you can always have it here and than put it into ssl-config and remove it from here. This is internal code, so it should not matter right?